src/main/java/org/broadinstitute/hellbender/engine/dataflow/transforms/RefBasesFromAPI.java

```diff
@@ -0,0 +1,65 @@
+package org.broadinstitute.hellbender.engine.dataflow.transforms;
+
+import com.google.cloud.dataflow.sdk.transforms.Create;
+import com.google.cloud.dataflow.sdk.transforms.DoFn;
+import com.google.cloud.dataflow.sdk.transforms.ParDo;
+import com.google.cloud.dataflow.sdk.transforms.View;
+import com.google.cloud.dataflow.sdk.values.KV;
+import com.google.cloud.dataflow.sdk.values.PCollection;
+import com.google.cloud.dataflow.sdk.values.PCollectionView;
+import org.broadinstitute.hellbender.engine.dataflow.datasources.RefAPIMetadata;
+import org.broadinstitute.hellbender.engine.dataflow.datasources.RefAPISource;
+import org.broadinstitute.hellbender.engine.dataflow.datasources.ReferenceShard;
+import org.broadinstitute.hellbender.utils.SimpleInterval;
+import org.broadinstitute.hellbender.utils.read.GATKRead;
+import org.broadinstitute.hellbender.utils.reference.ReferenceBases;
+
+/**
+ * RefBasesFromAPI queries the Google Genomics API for reference bases that span all of the reads on each shard.
+ *
+ * |---------------- shard 0 ----------------|---------------- shard 0 ----------------|
+ * |-- read a --|   |---- read b ---|           |--------------- read c --------------|
+ *
+ * results in,
+ * |---------- ref bases 1 ---------|           |---------- ref bases 2 --------------|
+ * |-- read a --|   |---- read b ---|           |--------------- read c --------------|
+ *
+ * KV<ref bases 1, [read a, read b]>
+ * KV<ref bases 2, [read c]>
+ */
+public class RefBasesFromAPI {
+    public static PCollection<KV<ReferenceBases, Iterable<GATKRead>>> getBasesForShard(PCollection<KV<ReferenceShard, Iterable<GATKRead>>> reads,
+                                                                                       RefAPISource refAPISource,
```

Comment:
I think you can get the same result by using a singleton for the RefApiSource. An added benefit is that it would probably result in more efficient code, because I'm not sure that Dataflow is smart enough to remove side inputs when the inputs are constant.